compiler/noirc_evaluator/src/ssa/opt/loop_invariant.rs

```diff
@@ -988,14 +988,18 @@ fn can_be_hoisted(
             }
         }
 
-        Constrain(..) | ConstrainNotEqual(..) | RangeCheck { .. } => hoist_with_predicate,
+        // A cast may have dependence on a range-check, which may not be hoisted, so we cannot always hoist a cast.
+        // Since at the end a cast is a no-op, not hoisting a cast should not have much impact.
```

Comment:
I don't think this is necessarily true, as a Cast not being hoisted could lead to downstream instructions not being hoisted. 

Could we add a small unit test that shows a cast dependent on a range check?